src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,105 @@
+     *
+     * @param input The user input containing the todo command.
+     * @return The description of the todo task.
+     * @throws DukeException If the description is empty.
+     */
+    public static String validateToDoCommand(String input) throws DukeException {
+        String[] inputArr = input.split(" ");
+        if (inputArr.length == 1) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        return input.replace("todo", "");
+    }
+
+    /**
+     * Validates and extracts the description and deadline from a deadline command.
+     *
+     * @param input The user input containing the deadline command.
+     * @return The description of the deadline task.
+     * @throws DukeException If the format is invalid or the description is empty.
+     */
+    public static String validateDeadlineCommand(String input) throws DukeException {
+        String[] inputArr = input.split(" ");
+        if (inputArr.length == 1) {
+            throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+        }
+
+        String[] description = input.replace("deadline", "").trim().split(" ");
+        int index = -1;
+        if (description.length > 2) {
+            for (int i = 0; i < description.length; i++) {
+                if (description[i].equals("/by")) {
+                    index = i;
+                    break;
+                }
+            }
+        }
+
+        if (index == -1 || index == 0 || index == description.length - 1) {
+            throw new DukeException("☹ OOPS!!! The format of a deadline is invalid. "
+                    + "Format: deadline <task name> /by <date>");
+        }
+
+        return input.replace("deadline", "");
+    }
+
+    /**
+     * Validates and extracts the description and event timing from an event command.
+     *
+     * @param input The user input containing the event command.
+     * @return The description of the event task.
+     * @throws DukeException If the format is invalid or the description is empty.
+     */
+    public static String validateEventCommand(String input) throws DukeException {
+        String[] inputArr = input.split(" ");
+        if (inputArr.length == 1) {
+            throw new DukeException("☹ OOPS!!! The description of a event cannot be empty.");
+        }
+
+        String[] description = input.replace("event", "").trim().split(" ");
+        int fromIndex = -1;
+        int toIndex = -1;
+        if (description.length > 2) {
+            for (int i = 0; i < description.length; i++) {
+                if (description[i].equals("/from")) {
+                    fromIndex = i;
+                }
+                if (description[i].equals("/to")) {
+                    toIndex = i;
+                }
+            }
+        }
+
+        if (fromIndex == -1 || toIndex == -1 || fromIndex == 0 || toIndex == 0 || fromIndex >= toIndex
```

Comment:
I'm not sure I like the large number of conditions. Is there a way to condense them or abstract them so that the code would flow more naturally?